src/dmd/dmodule.d

```diff
@@ -1319,6 +1319,22 @@ extern (C++) final class Module : Package
     {
         v.visit(this);
     }
+
+    /***********************************************
+     * Writes this module fully-qualified name to buf
+     * Params:
+     *    buf = The buffer to write to
+     */
+    void fullyQualifiedName(ref OutBuffer buf)
+    {
+        buf.writestring(ident.toString());
+
+        for (auto package_ = parent; package_ !is null; package_ = package_.parent)
+        {
+            buf.prependstring(".");
+            buf.prependstring(package_.ident.toChars());
```

Comment:
`.toString()` should work here, and is much more efficient

Reply:
`prependstring` takes a `const(char)*`. I _could_ call `toString` and pass its pointer, but I don't see why nor do I understand why `toString` would be more efficient since the implementations of `toString` and `toChars` are nearly identical:

```d
    override const(char)* toChars() const pure
    {
        return string;
    }

    extern (D) final const(char)[] toString() const pure
    {
        return string[0 .. len];
    }
```